File: WG_Repository/ViewModels/VMTenant.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using WishGrid.ViewModels.Shared;

namespace WishGrid.ViewModels
{
    public class VMTenant : VMBase
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public string TenantURL { get; set; }

        public bool Moderation { get; set; }
    }
    public class VMTenantEdit : VMTenant
    {
        public int LoggedUserId { get; set; }
        public int LoggedUserRole { get; set; }
    }

    public class VMTenantImage : VMTenant
    {
        public string URLImage { get; set; }
        public string URLTenant { get; set; }
    }
}

File: WGViewA5/src/app/_models/VMTenant.ts
import { VMUserSimple } from "./VMUser";

export interface VMTenant {
  id?: number;
  tenantURL?: string;
  moderation?: boolean;
}

export interface VMTenantEdit extends VMTenant {
  loggedUserId: number;
  loggedUserRole: number;
}

export interface VMTenantImage extends VMTenant
{
  urlImage: string;
  urlTenant: string;
}
